### ISSUE TITLE
[User Feature Request] More Flexible LR Strategy

### ISSUE THREAD

**Issue by lispc** [PaddlePaddle/Paddle#3721]
There are many publications that use a very flexible LR strategy(eg: [Accurate, Large Minibatch SGD: Training ImageNet in 1 Hour](https://arxiv.org/abs/1706.02677), and [Super-Convergence: Very Fast Training of Residual Networks Using Large Learning Rates](https://arxiv.org/abs/1708.07120)), while the lr strategies of  PaddlePaddle cannot satisfy these requirements. So from a user's perspective, it will be very useful for PaddlePaddle that  applications can **change lr manually after every minbatch**.       
In Tensorflow, this can be accomplished by set lr as feed_dict, and in PyTorch,  users can directly assign the internal lr variable.

**Comment by qingqing01** [comment]
@lispc Actually, PaddlePaddle support to set learning rate manually, see the comments in `Proto`: https://github.com/PaddlePaddle/Paddle/blob/develop/proto/TrainerConfig.proto#L126  and code : [ManualLRS ](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/parameter/LearningRateScheduler.cpp#L115) and [PassManualLRS](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/parameter/LearningRateScheduler.cpp#L163) .

But we are lack of using documents.  :(
